### PR TITLE
[nginxreceiver] Updated for upcoming client auth extension changes

### PR DESCRIPTION
**Pull request by gramidt**
**Description:** 
Updated nginxreceiver for upcoming client auth extension changes

**Link to tracking Issue:**
#3507 

**Testing:**
Unit tests

**Documentation:**
Unit tests

### CODE REVIEW

File: receiver/nginxreceiver/scraper.go [hunk 1]
```diff
@@ -43,16 +46,23 @@ func newNginxScraper(
...
+	if err != nil {
+		return err
+	}
+
+	r.httpClient = httpClient
+
+	return nil
+}
+
+func (r *nginxScraper) scrape(_ context.Context) (pdata.ResourceMetricsSlice, error) {
```
Comment (bogdandrutu) [inline receiver/nginxreceiver/scraper.go:60]: s/scrape(_ context.Context)/scrape(context.Context)

File: receiver/nginxreceiver/scraper.go [hunk 2]
```diff
@@ -43,16 +46,23 @@ func newNginxScraper(
...
 
-func (r *nginxScraper) scrape(ctx context.Context) (pdata.ResourceMetricsSlice, error) {
+func (r *nginxScraper) start(_ context.Context, _ component.Host) error {
+	httpClient, err := r.cfg.ToClient()
+	if err != nil {
+		return err
+	}
+
+	r.httpClient = httpClient
+
```
Comment (bogdandrutu) [inline receiver/nginxreceiver/scraper.go:56]: Unnecessary empty line.

File: receiver/nginxreceiver/scraper_test.go [hunk 3]
```diff
@@ -123,7 +128,24 @@ func TestScraperError(t *testing.T) {
...
 		})
-		_, err := sc.scrape(context.Background())
+		err := sc.start(context.Background(), componenttest.NewNopHost())
+		require.NoError(t, err)
+		_, err = sc.scrape(context.Background())
 		require.Equal(t, errors.New("failed to parse response body \"Bad status page\": invalid input \"Bad status page\""), err)
 	})
 }
+
+func TestScraperStart(t *testing.T) {
```
Comment (bogdandrutu) [inline receiver/nginxreceiver/scraper_test.go:138]: s/TestScraperStart/TestScraperFailedStart

**Comment by tigrannajaryan** [comment]
@keitwb please review. (Also can you please add yourself as a codeowner for nginxreceiver).

**Review by keitwb — APPROVED** [review]
LGTM

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**